### PR TITLE
add new features(getFirstDayOfWeek , getFirstDayOfMonth , getFirstDayOfYear)

### PR DESCRIPTION
**Pull request by amirfaramarzi**

### CODE REVIEW

**Comment by morilog** [comment]
Please write unit tests for these functions

File: tests/JalalianTest.php [hunk 1]
```diff
@@ -129,4 +129,37 @@ public function testGetWeekOfMonth()
         $jDate = new Jalalian(1395, 7, 20);
         $this->assertEquals($jDate->getWeekOfMonth(), 4);
     }
+    
+    public function testGetFirstDayOfWeek()
+    {
+        $jDate = new Jalalian(1401, 1, 23);
+        $this->assertEquals($jDate->getFirstDayOfWeek()->format('Y-m-d'), '1401-01-20');
+
+        for ($i=0;$i<5;$i++){
```
Comment (morilog) [inline tests/JalalianTest.php:138]: Please fix code style

File: tests/JalalianTest.php [hunk 2]
```diff
@@ -129,4 +129,37 @@ public function testGetWeekOfMonth()
...
+            $this->assertEquals($jDate->getFirstDayOfWeek()->format('Y-m-d'), $jDate->subDays($jDate->getDayOfWeek())->format('Y-m-d'));
+        }
+    }
+
+    public function testGetFirstDayOfMonth()
+    {
+        $jDate = new Jalalian(1401, 1, 23);
+        $this->assertEquals($jDate->getFirstDayOfMonth()->format('Y-m-d'), '1401-01-01');
+
+        for ($i=0;$i<5;$i++){
```
Comment (morilog) [inline tests/JalalianTest.php:149]: Please fix code style here too.

**Comment by amirfaramarzi** [comment]
please check again!

**Pull request merged**